Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -303,6 +302,37 @@ extension CoreDataManager {
     }
 }
 
+// MARK: Helpers
+private extension CoreDataManager {
+
+    func saveStorageIfNeeded(_ storage: StorageType) {
+        let context = storage as! NSManagedObjectContext
+        guard context.hasChanges else {
+            return
+        }
+        do {
+            try context.save()
```

Comment:
👋 @itsmeichigo , Does calling `context.save()` mean we no longer call the `saveIfNeeded()` from `NSManagedObjectContext+Storage` file? We rely on that method to enforce correct context usage for write operations, right? Please correct me if I am missing something.

Reply:
Hi @selanthiraiyan - since my last change in the Core Data stack, I don't expect anyone to use `saveIfNeeded` directly, but to use `performAndSave` instead - which handles the write operations on the background thread.

`saveIfNeeded` was only kept because we have a lot of unit tests relying on it - I even marked it as deprecated in `MockStorageManager`.

Reply:
Thanks, @itsmeichigo!

Sorry, I am trying to understand this. I raised this question because in `performAndSave` method the following [lines 1](https://github.com/woocommerce/woocommerce-ios/pull/14961/files#diff-f43a023c555398d48483bd4415451807f83ae569c2b166696c6bfc8c1fee14e6L114-L115) and [lines 2](https://github.com/woocommerce/woocommerce-ios/pull/14961/files#diff-f43a023c555398d48483bd4415451807f83ae569c2b166696c6bfc8c1fee14e6L141) were using `saveIfNeeded()` before we started calling `saveStorageIfNeeded()` instead. 🤔 

Did we miss replacing those `saveIfNeeded()` calls from `performAndSave` method in the past while planning to deprecate the method?

Reply:
I wanted to deprecate **direct access** to that method in tests and use `performAndSave` to ensure the correct thread usage. Now that we use the new method to handle saving inside of `performAndSave`, we can remove the method. I added a commit for that here: 122e1c6ae59bb446824701ef6f792327aa2c35c9. Let me know if you have any other questions! @selanthiraiyan